### PR TITLE
Pal 145 move data services into services repo

### PR DESCRIPTION
**Pull request by nw1984**

### CODE REVIEW

File: data-service/pom.xml [hunk 1]
```diff
@@ -28,5 +28,311 @@
...
+            <artifactId>spring-cloud-starter-openfeign</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-config-client</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-starter-ribbon</artifactId>
+            <version>1.4.7.RELEASE</version>
```
Comment (m09526) [inline data-service/pom.xml]: Remove version number

File: data-service/pom.xml [hunk 2]
```diff
@@ -28,5 +28,311 @@
 
     <artifactId>data-service</artifactId>
 
+    <dependencies>
+        <dependency>
```
Comment (m09526) [inline data-service/pom.xml:37]: Is this dependency needed?

File: data-service/pom.xml [hunk 3]
```diff
@@ -28,5 +28,311 @@
...
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.gchq.palisade</groupId>
+            <artifactId>common</artifactId>
+            <version>${parent.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.gchq.palisade</groupId>
+            <artifactId>readers-common</artifactId>
+            <version>0.4.1-SNAPSHOT</version>
```
Comment (m09526) [inline data-service/pom.xml]: Change version to use ${project.parent.version}

File: data-service/pom.xml [hunk 4]
```diff
@@ -28,5 +28,311 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.gchq.palisade</groupId>
+            <artifactId>common</artifactId>
+            <version>${parent.version}</version>
```
Comment (m09526) [inline data-service/pom.xml]: Change version to use ${project.parent.version}

File: data-service/pom.xml [hunk 5]
```diff
@@ -28,5 +28,311 @@
...
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.gchq.palisade</groupId>
+            <artifactId>readers-common</artifactId>
+            <version>0.4.1-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.gchq.palisade</groupId>
+            <artifactId>hadoop-readers</artifactId>
+            <version>0.4.1-SNAPSHOT</version>
```
Comment (m09526) [inline data-service/pom.xml]: Change version to use ${project.parent.version}

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/web/DataServiceController.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @PostMapping(value = "/read", consumes = "application/json", produces = "application/json")
+    public ReadResponse readSync(@RequestBody final ReadRequest request) {
+        LOGGER.debug("Invoking read: {}", request);
+        return read(request).join();
+    }
+
+    // Taken from the following example: https://dzone.com/articles/streaming-data-with-spring-boot-restful-web-service
+    @PostMapping(value = "/read/chunked", consumes = "application/json", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (m09526) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/web/DataServiceController.java]: Is this produces parameter right? This sends back an octet stream.

**Review by m09526 — CHANGES_REQUESTED** [review]
Nice changes. Minor mods needed.

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    @Bean
+    public CacheConfiguration cacheConfiguration() {
+        return new CacheConfiguration();
+    }
+
+    @Bean
+    public SimpleDataService SimpleDataService(final Map<String, BackingStore> backingStores,
+                                               final AuditClient auditClient,
+                                               final PalisadeClient palisadeClient) {
+        AuditService audService = auditService(auditClient);
```
Comment (dev-958) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java]: No! You already have a bean of type AuditService, inject that.

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(ApplicationConfiguration.class);
+
+    @Bean
+    public CacheConfiguration cacheConfiguration() {
+        return new CacheConfiguration();
+    }
+
+    @Bean
+    public SimpleDataService SimpleDataService(final Map<String, BackingStore> backingStores,
+                                               final AuditClient auditClient,
```
Comment (dev-958) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java]: inject PalisadeService

Comment (dev-958) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java]: inject AuditRequestReceiver

**Review by dev-958 — CHANGES_REQUESTED** [review]
See comments

File: data-service/pom.xml (hunk 1 above)
Comment (nw1984) [reply data-service/pom.xml]: spring-cloud-starter-parent is a starter parent that does not follow the normal spring cloud starter naming convention i.e it does not follow that standard bill of materials format.
I have added in a specific property for this.

**Review by m09526 — DISMISSED** [review]

**Review by dev-958 — DISMISSED** [review]

File: audit-service/maven_output.txt [hunk 9]
```diff
@@ -0,0 +1,3188 @@
+Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T20:00:29+01:00)
```
Comment (m09526) [inline audit-service/maven_output.txt]: This file is immense and also contains personal data like names in the file path. This shouldn't be here.

**Review by m09526 — CHANGES_REQUESTED** [review]

**Pull request merged**